dirconfig/dirconfig_test.go

```diff
@@ -0,0 +1,262 @@
+	"time"
+
+	"github.com/project-koku/korekuta-operator-go/testlogr"
+)
+
+type MockFileInfo struct {
+	name  string
+	isDir bool
+}
+
+func NewMockFileInfo(name string, isDir bool) MockFileInfo {
+	return MockFileInfo{
+		name:  name,
+		isDir: isDir,
+	}
+}
+
+func (mfi MockFileInfo) Name() string {
+	return mfi.name
+}
+
+func (mfi MockFileInfo) Size() int64 {
+	return 100
+}
+
+func (mfi MockFileInfo) Mode() os.FileMode {
+	if mfi.isDir {
+		return os.ModeDir
+	}
+	return os.ModeAppend
+}
+
+func (mfi MockFileInfo) ModTime() time.Time {
+	return time.Unix(110, 0)
+}
+
+func (mfi MockFileInfo) IsDir() bool {
+	return mfi.isDir
+}
+
+func (mfi MockFileInfo) Sys() interface{} {
+	return nil
+}
+
+func TestDirString(t *testing.T) {
+	tcs := []struct {
+		path     string
+		expected string
+	}{
+		{"/tmp/dir/cfg", "/tmp/dir/cfg"},
+		{"/etc/data", "/etc/data"},
+		{"/root/configs/costmanagement", "/root/configs/costmanagement"},
+	}
+
+	for _, tc := range tcs {
+		dir := &Directory{Path: tc.path}
+		if dir.String() != tc.expected {
+			t.Errorf("Expected dir.String() to return %v but got %v", tc.expected, dir.String())
+		}
+	}
+}
+
+var listDirFileMock = func(files []os.FileInfo, err error) DirListFunc {
+	return func(path string) ([]os.FileInfo, error) {
+		return files, err
+	}
+}
+var removeAllMock = func(err error) RemoveAllFunc {
+	return func(path string) error {
+		return err
+	}
+}
+
+var statMock = func(err error) StatFunc {
+	return func(path string) (os.FileInfo, error) {
+		return nil, err
+	}
+}
+
+var createDirMock = func(err error) DirCreateFunc {
+	return func(path string) error {
+		return err
+	}
+}
+
+func TestDirRemoveContents(t *testing.T) {
+	dirPath := "/bla/configs/"
+	tcs := []struct {
+		listDirFile   DirListFunc
+		removeAll     RemoveAllFunc
+		expectedError error
+	}{
+		{
+			listDirFileMock(nil, fmt.Errorf("Oh no!")),
+			removeAllMock(nil),
+			fmt.Errorf("RemoveContents: could not read directory: Oh no!"),
+		},
+		{
+			listDirFileMock([]os.FileInfo{}, nil),
+			removeAllMock(nil),
+			nil,
+		},
+		{
+			listDirFileMock([]os.FileInfo{NewMockFileInfo("/tmp/dir/cfg", false)}, nil),
+			removeAllMock(fmt.Errorf("oops")),
+			fmt.Errorf("RemoveContents: could not remove file: oops"),
+		},
+	}
+
+	for _, tc := range tcs {
+		dir := &Directory{
+			Path: dirPath,
+			DirectoryFileSystem: &DirectoryFileSystem{
+				ListDirectory:   tc.listDirFile,
+				RemoveAll:       tc.removeAll,
+				Stat:            statMock(nil),
+				CreateDirectory: createDirMock(nil),
+			},
+		}
+		err := dir.RemoveContents()
+		if tc.expectedError != nil || err != nil {
+			if (tc.expectedError.Error() != err.Error()) || (tc.expectedError != nil && err == nil) || (tc.expectedError == nil && err != nil) {
+				t.Errorf("Expected to return error: %v but got %v", tc.expectedError, err)
+			}
+		}
+	}
+}
+
+func TestDirExists(t *testing.T) {
+	dirPath := "/tmp/blalal23392910302919304029191001111fake-non-exist-void-234321123455fd344fdhshjka"
```

Comment:
Let's make this string random every time the test runs. You can use this helper to create random strings:
[https://github.com/project-koku/korekuta-operator-go/blob/master/collector/suite_test.go#L155](https://github.com/project-koku/korekuta-operator-go/blob/master/collector/suite_test.go#L155)